feelslikemonday/app/src/main/java/com/example/feelslikemonday/ui/map/MapsActivity.java

```diff
@@ -0,0 +1,128 @@
+import com.google.android.gms.maps.CameraUpdateFactory;
+import com.google.android.gms.maps.GoogleMap;
+import com.google.android.gms.maps.OnMapReadyCallback;
+import com.google.android.gms.maps.SupportMapFragment;
+import com.google.android.gms.maps.model.BitmapDescriptor;
+import com.google.android.gms.maps.model.BitmapDescriptorFactory;
+import com.google.android.gms.maps.model.LatLng;
+import com.google.android.gms.maps.model.MarkerOptions;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.google.android.gms.common.internal.safeparcel.SafeParcelable.NULL;
+
+public class MapsActivity extends FragmentActivity implements OnMapReadyCallback {
+
+    private GoogleMap mMap;
+    private SharedPreferences pref;
+    private String myUserID;
+    private User currentUser;
+    private ArrayList<MoodEvent> myEmotionList = new ArrayList<>();
+    private List<MoodEvent> myCurrentMoodList;
+    private LatLng currentLocation;
+    private BitmapDescriptor markerType;
+
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_maps);
+        // Obtain the SupportMapFragment and get notified when the map is ready to be used.
+        SupportMapFragment mapFragment = (SupportMapFragment) getSupportFragmentManager()
+                .findFragmentById(R.id.map);
+        mapFragment.getMapAsync(this);
+    }
+
+
+    /**
+     * Manipulates the map once available.
+     * This callback is triggered when the map is ready to be used.
+     * This is where we can add markers or lines, add listeners or move the camera.
+     * If Google Play services is not installed on the device, the user will be prompted to install
+     * it inside the SupportMapFragment. This method will only be triggered once the user has
+     * installed Google Play services and returned to the app.
+     */
+
+    @Override
+    public void onMapReady(GoogleMap googleMap) {
+        mMap = googleMap;
+
+        pref = getApplicationContext().getSharedPreferences(SignupActivity.PREFS_NAME, 0);
+        myUserID = pref.getString(SignupActivity.USERNAME_KEY, null);
+
+        UserDAO userDAO = new UserDAO();
+        userDAO.get(myUserID, new UserCallback() {
+            @Override
+            public void onCallback(User user) {
+                currentUser = user;
+                myEmotionList.clear();
+                myCurrentMoodList = currentUser.getMoodHistory();
+                for (int i = 0; i < myCurrentMoodList.size(); i++) {
+                    myEmotionList.add(myCurrentMoodList.get(i));
+                }
+                placeMarkers();
+            }
+        }, new VoidCallback() {
+            @Override
+            public void onCallback() {
+                Log.v("succc", "succ");
+            }
+        });
+    }
+
+    private void placeMarkers() {
+        for (int i = 0; i < myEmotionList.size(); i++) {
```

Comment:
You might already know this, but you can also write this as `for (EmotionType emotion: myEmotionList)` to save a bit of time typing and reading. This implementation fine too, and probably can be left as is